persistence/db.go

```diff
@@ -25,34 +22,31 @@ func init() {
 	rand.Seed(time.Now().UnixNano())
 }
 
-var _ modules.PersistenceContext = &PostgresContext{}
+var _ modules.PersistenceRWContext = &PostgresContext{}
 
 // TODO: These are only externalized for testing purposes, so they should be made private and
 //       it is trivial to create a helper to initial a context with some values.
 type PostgresContext struct {
```

Comment:
PostgresDB is just Tx and Blockstore, so if you'd prefer to remove that level of abstraction, it would look like this:

```
type PostgresContext struct {
	Height int64
	Tx         pgx.Tx
	Blockstore kvstore.KVStore	
}
```

Not

```
type PostgresContext struct {
	Height int64
	DB     PostgresDB
	Tx         pgx.Tx
	Blockstore kvstore.KVStore	
}
```

Reply:
If you'd like this done, let's push it to #149

Reply:
Sgtm. Added this as an explicit deliverable:

![Screen Shot 2022-08-11 at 1 27 18 PM](https://user-images.githubusercontent.com/1892194/184195882-70534f20-759c-44bf-8e55-8c1affbdabc8.png)